--- lab.c_old	2023-11-21 01:20:45.229613756 +0300
+++ lab.c	2023-11-21 01:20:49.965642138 +0300
@@ -32,6 +32,17 @@
 
     srand(randseed);
 
+    if (argc < 2) {
+        fprintf(stderr, "Usage: %s <size>\n", argv[0]);
+        return 1;
+    }
+
+    size = atoi(argv[1]);
+    if (size < 1) {
+        fprintf(stderr, "Invalid size: %d (must be nonnegative)\n", size);
+        return 1;
+    }
+
     char *hpaths = calloc((size-1)*size, sizeof(char));
     char *vpaths = calloc((size-1)*size, sizeof(char));
     char *visited = calloc(size*size, sizeof(char));
