--- lab.c_old	2023-11-21 01:20:45.229613756 +0300
+++ lab.c	2023-11-21 01:24:43.923130700 +0300
@@ -1,6 +1,7 @@
 #include <stdio.h>
 #include <time.h>
 #include <stdlib.h>
+#include <string.h>
 
 #define DIRECTION_NUM 4
 
@@ -30,8 +31,28 @@
     char cEmpty = '.';
     unsigned randseed = time(NULL);
 
-    srand(randseed);
+    if (argc < 4) {
+        fprintf(stderr, "Usage: %s <seed> <chars> <size>\n", argv[0]);
+        return 1;
+    }
+
+    randseed = atoi(argv[1]);
+
+    char* chars = argv[2];
+    if (strlen(chars) != 2) {
+        fprintf(stderr, "<chars> must have exactly two characters!\n");
+        return 1;
+    }
+    cEmpty = chars[0];
+    cWall = chars[1];
+
+    size = atoi(argv[3]);
+    if (size < 1) {
+        fprintf(stderr, "Invalid size: %d (must be nonnegative)\n", size);
+        return 1;
+    }
 
+    srand(randseed);
     char *hpaths = calloc((size-1)*size, sizeof(char));
     char *vpaths = calloc((size-1)*size, sizeof(char));
     char *visited = calloc(size*size, sizeof(char));
